presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoPreparedStatement.java

```diff
@@ -33,128 +36,163 @@
     {
-        throw new NotImplementedException("PreparedStatement", "setShort");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatLongLiteral(x), Types.SMALLINT));
     }
 
     @Override
     public void setInt(int parameterIndex, int x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setInt");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatLongLiteral(x), Types.INTEGER));
     }
 
     @Override
     public void setLong(int parameterIndex, long x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setLong");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatLongLiteral(x), Types.BIGINT));
     }
 
     @Override
     public void setFloat(int parameterIndex, float x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setFloat");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatDoubleLiteral(x), Types.REAL));
     }
 
     @Override
     public void setDouble(int parameterIndex, double x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setDouble");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatDoubleLiteral(x), Types.DOUBLE));
     }
 
     @Override
     public void setBigDecimal(int parameterIndex, BigDecimal x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setBigDecimal");
+        checkOpen();
+        setParameter(parameterIndex, formatDecimalLiteral(x));
     }
 
     @Override
     public void setString(int parameterIndex, String x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setString");
+        checkOpen();
+        setParameter(parameterIndex, formatStringLiteral(x));
     }
 
     @Override
     public void setBytes(int parameterIndex, byte[] x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setBytes");
+        checkOpen();
+        setParameter(parameterIndex, formatBinaryLiteral(x));
     }
 
     @Override
     public void setDate(int parameterIndex, Date x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setDate");
+        checkOpen();
+        setParameter(parameterIndex, formatDateLiteral(DATE_FORMATTER.print(x.getTime())));
     }
 
     @Override
     public void setTime(int parameterIndex, Time x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setTime");
+        checkOpen();
+        setParameter(parameterIndex, formatTimeLiteral(TIME_FORMATTER.print(x.getTime())));
     }
 
     @Override
     public void setTimestamp(int parameterIndex, Timestamp x)
```

Comment:
Are you going to support `java.sql.PreparedStatement#setTimestamp(int, java.sql.Timestamp, java.util.Calendar)` too?

Reply:
I'd need to research how this should work. Will leave this for the future.